tools/test_util.py

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env python
+# Copyright 2018-2019 the Deno authors. All rights reserved. MIT license.
+# Runs the full test suite.
+# Usage: ./tools/test.py out/Debug
+import argparse
+import os
+import sys
+import unittest
+
+from util import (enable_ansi_colors, build_path, RESET, FG_RED, FG_GREEN,
+                  executable_suffix, run, run_output, rmtree, tests_path)
+
+
+class DenoTestCase(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        args = parse_test_args()
+
+        cls.build_dir = args.build_dir
+        cls.deno_exe = args.executable
+
+
+# overload the test result class
+class ColorTextTestResult(unittest.TextTestResult):
+    def getDescription(self, test):
+        name = str(test)
+        if name.startswith("test_"):
+            name = name[5:]
+        return name
+
+    def addSuccess(self, test):
+        if self.showAll:
+            self.stream.write(FG_GREEN)
+        super(ColorTextTestResult, self).addSuccess(test)
+        if self.showAll:
+            self.stream.write(RESET)
+
+    def addError(self, test, err):
+        if self.showAll:
+            self.stream.write(FG_RED)
+        super(ColorTextTestResult, self).addError(test, err)
+        if self.showAll:
+            self.stream.write(RESET)
+
+    def addFailure(self, test, err):
+        if self.showAll:
+            self.stream.write(FG_RED)
+        super(ColorTextTestResult, self).addFailure(test, err)
+        if self.showAll:
+            self.stream.write(RESET)
+
+
+class ColorTextTestRunner(unittest.TextTestRunner):
+    resultclass = ColorTextTestResult
+
+
+def create_test_arg_parser():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        '--failfast', '-f', action='store_true', help='Stop on first failure')
+    parser.add_argument(
+        '--verbosity', '-v', action='store_true', help='Verbose output')
+    parser.add_argument("--executable", help="Use external executable of Deno")
```

Comment:
Should we make --deno-dir a kwarg too (rather than position)? I think that's clearer.

Also, I guess required=False below is not required (my mistake).

Reply:
I removed `required=False` as for `--deno-dir` what would it do?

Reply:
Atm it's the position argument. See code below

Reply:
@hayd `--deno-dir` in place of `build_dir` might be confusing, they're basically two different things. I'm +1 for `--build-dir` instead of positional argument, but let's not mix the names.